theano/tensor/nnet/tests/test_neighbours.py

```diff
@@ -366,6 +366,37 @@ def speed_neibs_wrap_centered(self):
         for i in range(1000):
             f()
 
+    def test_infer_shape(self):
+        shape = (100, 40, 6, 6)
+        images = numpy.ones(shape).astype('float32')
+        x = T.ftensor4()
+        f = self._compile_and_check([x],
+                                    [images2neibs(
+                                        x, neib_shape=(2,2),
+                                        mode='valid')],
+                                    [images],
+                                    Images2Neibs
+                                    )
+        shape = (100, 40, 5, 5)
```

Comment:
Test ignore border with this shape (100, 40, 6, 6). This is another corner case.